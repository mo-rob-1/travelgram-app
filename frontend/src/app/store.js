import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../features/auth/authSlice";
import imageReducer from "../features/images/imageSlice";

export const store = configureStore({
  reducer: {
    auth: authReducer,
    images: imageReducer,
  },
});
