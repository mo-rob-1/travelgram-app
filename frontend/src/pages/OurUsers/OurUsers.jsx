import { useEffect, useState } from "react";
import moment from "moment";
import {
  List,
  ListItem,
  ImageWrapper,
  Avatar,
  Title,
  StyledCardLink,
  UsernameTitle,
  UserTextWrapper,
  ViewProfile,
  Section,
} from "./OurUsers.styled";
import axios from "axios";

function OurUsers() {
  const [users, setUsers] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    axios
      .get("/api/users")
      .then((res) => {
        setUsers(res.data.reverse());
        setLoading(false);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  if (loading) {
    return (
      <Section>
        <h2>Loading...</h2>
      </Section>
    );
  }

  return (
    <Section>
      <Title>Our Users</Title>
      <List data-testid="list">
        {users.map((user) => (
          <ListItem key={user.cloudinary_id}>
            <StyledCardLink to={`/our-users/${user._id}`}>
              <ImageWrapper>
                <Avatar src={user.avatar} alt={user.username} />
              </ImageWrapper>

              <UserTextWrapper>
                <UsernameTitle>{user.username}</UsernameTitle>
                <div style={{ marginTop: ".7rem", marginBottom: ".7rem", display: "grid", gap: ".4rem" }}>
                  <p>{user.name}</p>
                  <p>{user.email}</p>
                  <p>Joined: {moment(user.createdAt).format("MMMM YYYY")}</p>
                </div>
                <ViewProfile to={`/our-users/${user._id}`}>View Profile</ViewProfile>
              </UserTextWrapper>
            </StyledCardLink>
          </ListItem>
        ))}
      </List>
    </Section>
  );
}

export default OurUsers;
