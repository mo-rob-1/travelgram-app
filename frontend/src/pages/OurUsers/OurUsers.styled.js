import styled from "styled-components";
import { Link } from "react-router-dom";

export const List = styled.ul`
  list-style-type: none;
  display: grid;
  gap: 1.5rem;
  text-align: center;
  @media (min-width: 768px) {
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: 1fr;
    margin-top: 2rem;
  }
  @media (min-width: 1024px) {
    grid-template-columns: repeat(3, 1fr);
  }
`;

export const ListItem = styled.li`
  background-color: #fff;
  padding-top: 2rem;
  padding-bottom: 2rem;
  border-radius: 10px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 3px 0px, rgba(0, 0, 0, 0.06) 0px 1px 2px 0px;
`;

export const ImageWrapper = styled.div`
  display: flex;
  justify-content: center;
`;

export const Avatar = styled.img`
  height: 100px;
  object-fit: contain;
  border-radius: 50%;
  width: 100px;
  object-fit: cover;
`;

export const Title = styled.h1`
  /* text-align: center; */
  @media (min-width: 768px) {
    margin-top: 1rem;
    /* text-align: left; */
  }
`;

export const StyledCardLink = styled(Link)`
  text-decoration: none;
  color: #222;
`;

export const UsernameTitle = styled.h3`
  display: block;
`;

export const UserTextWrapper = styled.div`
  display: grid;
  gap: 0.5rem;
  margin-top: 1rem;
`;

export const ViewProfile = styled(Link)`
  color: #222;
  margin-top: 1rem;
  font-weight: bold;
`;

export const Section = styled.section`
  margin-top: 1rem;

  @media (min-width: 1024px) {
    margin-top: 3rem;
  }
`;
