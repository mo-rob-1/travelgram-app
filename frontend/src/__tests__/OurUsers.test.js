import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";
import OurUsers from "../pages/OurUsers/OurUsers";

describe("OurUsers", () => {
  it("renders the our users page", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <OurUsers />
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the list of users", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <OurUsers />
        </Router>
      </Provider>
    );

    const list = getByTestId("list");
    expect(list).toBeTruthy();
  });
});
