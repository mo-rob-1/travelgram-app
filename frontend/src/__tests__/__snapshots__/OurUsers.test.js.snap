// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OurUsers renders the our users page 1`] = `
<DocumentFragment>
  <section
    class="sc-papXJ jJMshX"
  >
    <h2>
      Loading...
    </h2>
  </section>
</DocumentFragment>
`;
